.github/workflows/docker-push.yaml

```diff
@@ -34,7 +32,10 @@ jobs:
       env:
         REGISTRY_NAMESPACE: ${{ github.event.inputs.REGISTRY_NAMESPACE }}
         IMAGE_TAG: ${{ github.event.inputs.IMAGE_TAG }}
+        IMAGE_TAG_SHA: ${{ github.ref_name }}-${{ github.sha }}
       run: |
-        make docker-build docker-push
+        # Set IMAGE_TAG to a calculated <branch>-<sha> if not specified
+        export IMAGE_TAG=${IMAGE_TAG:-${IMAGE_TAG_SHA:0:-33}}
```

Comment:
if the TAG is not provided can we push with branch name or use the latest for main branch and branch name for the released branch, this ensures that we don't push to many images in long term

Reply:
@Madhu-1 This action will not push images automatically, It will only push if someone invokes it. So we should be good.

Reply:
i still like to change that as we are making the TAG as optional in this PR.

Reply:
@Madhu-1 What exactly are you suggesting, if we put a specific tag it will become mandatory

Reply:
instead of Hash/SHA i would vote to tag as on the branch name to avoid having many tags in the registry.

Reply:
@Madhu-1 We need unique images for the odf and ocs upstream ci, If we push the images with same tag we may see a lot of failures in the odf and ocs CI.

Reply:
can you please explain it a bit why we need to have unique names?

Reply:
I am not talking about the unique names, I am talking about the unique tags.

client-operator and ocs-operator are tightly coupled with each other. We should stick to one image unless the required changes have merged in both the repos together. Moreover, why are we worried about having a lot of images? That is how the whole industry is doing. for example rook.

Reply:
>why are we worried about having a lot of images? 

This will a concern about maintenance and cleanup why we need to have lot of images without a need is still the question.

sorry that was a typo, i meant unique tag. i don't see a reason to take example from the whole industry or Rook and try to impose it here without a proper reason. Even if they are tight-coupled, could you please explain why the tags need to be unique here?

Reply:
Lets say something has merged in the ocs-client-operator but we have not made the required changes to the ocs-operator. In this case ocs-operator CI will pick the latest changes from the ocs-client-operator and will fail the CI.

It will put us at risk and all the PR's will be blocked due to the failing CI. Moreover, it will take a lot of cycles to dig into the CI and look at failures.

Reply:
PR is still required, any resolution can be reached?

Reply:
Yes we do need a PR or someone has to manually create the images everytime

Reply:
okay, merging this one as its a manual workflow trigger not the automated one, but now its the admin/maintainer responsibility to ensure they set the required details and avoid pushing more images to the quay